kornia/onnx/sequential.py

```diff
@@ -0,0 +1,150 @@
+from typing import Optional, Union
+
+from kornia.core.external import numpy as np
+from kornia.core.external import onnx
+from kornia.core.external import onnxruntime as ort
+
+__all__ = ["ONNXSequential"]
+
+
+class ONNXSequential:
+    """ONNXSequential to chain multiple ONNX operators together.
+
+    Args:
+        *args:
+            A variable number of ONNX models (either ONNX ModelProto objects or file paths).
+        providers:
+            A list of execution providers for ONNXRuntime (e.g., ['CUDAExecutionProvider', 'CPUExecutionProvider']).
+        session_options:
+            Optional ONNXRuntime session options for optimizing the session.
+        io_maps:
+            An optional list of list of tuples specifying input-output mappings for combining models.
+            If None, we assume the default input name and output name are "input" and "output" accordingly, and
+            only one input and output node for each graph.
+            If not None, `io_maps[0]` shall represent the `io_map` for combining the first and second ONNX models.
+    """
+
+    def __init__(
+        self,
+        *args: Union[onnx.ModelProto, str],  # type:ignore
+        providers: Optional[list[str]] = None,
+        session_options: Optional[ort.SessionOptions] = None,  # type:ignore
+        io_maps: Optional[list[tuple[str, str]]] = None,
+    ) -> None:
+        self.operators = args
+        self._combined_op = self._combine(io_maps)
+        self._session = self.create_session()
+
+    def _load_op(self, arg: Union[onnx.ModelProto, str]) -> onnx.ModelProto:  # type:ignore
+        """Loads an ONNX model, either from a file path or use the provided ONNX ModelProto.
+
+        Args:
+            arg: Either an ONNX ModelProto object or a file path to an ONNX model.
+
+        Returns:
+            onnx.ModelProto: The loaded ONNX model.
+        """
+        if isinstance(arg, str):
+            return onnx.load(arg)  # type:ignore
+        return arg
+
+    def _combine(self, io_maps: Optional[list[tuple[str, str]]] = None) -> onnx.ModelProto:  # type:ignore
```

Comment:
This feature is very powerful to create high level pipeline but from the docs is a but unclear how it works and how you link the different models you provide in the constructor

Reply:
there is a `io_map` you need to define. with single one-to-one model, we assume the input node is "input" and output node is "output". If you got multiple input or output nodes, you gonna need to pass the `io_map` arg